client/src/pages/user-preferences/UserPreferences.js

```diff
@@ -0,0 +1,78 @@
+import { Link } from "react-router-dom";
+import Form from "react-bootstrap/Form";
+import { useForm } from "react-hook-form";
+import Data from "../../components/form-data/formdata";
+import "./UserPreferences.css";
+
+export default function UserPreferences() {
+	const {
+		register,
+		handleSubmit,
+		formState: { errors },
+	} = useForm();
+	const onSubmit = (data) => {
+		console.log(data);
+	};
+
+	return (
+		<Form className="form" onSubmit={handleSubmit(onSubmit)}>
+			<Form.Group className="form-group2" controlId="service">
+				<Form.Label>Please select the service:</Form.Label>
+				<Form.Select
+					aria-label="Default select example"
+					className="w-50"
+					type="text"
+					name="service"
+					{...register("service", {
+						required: "Please select your service",
+					})}
+                    placeholder="Select here..."
+                    onChange={(e) => console.log(e.target.value)}
+				>
+                    <option value="">Select here...</option>
+					{Data.services.map((service) => {
+						return (
+							<option key={service} value={service}>
+								{service}
+							</option>
+						);
+					})}
+				</Form.Select>
+                <Form.Text className="text-muted">
+                {errors.service && errors.service.message}
+                </Form.Text>
+			</Form.Group>
+			<Form.Group className="form-group2" controlId="location">
+				<Form.Label>Please select preferred location for service:</Form.Label>
+				<Form.Select
+					aria-label="Default select example"
+					className="w-50"
+					type="text"
+					name="location"
+					{...register("location", {
+						required: "Please select your location",
+					})}
+                    placeholder="Select here..."
+                    onChange={(e) => console.log(e.target.value)}
+				>
+                    <option value="">Select here...</option>
+					{Data.location.map((location) => {
+						return (
+							<option key={location} value={location}>
+								{location}
+							</option>
+						);
+					})}
+				</Form.Select>
+                <Form.Text className="text-muted">
+                {errors.location && errors.location.message}
+                </Form.Text>
+			</Form.Group>
+			<div className="container-btn">
+				<Link className="btn btn-primary" to="#" onClick={handleSubmit(onSubmit)}>
```

Comment:
1.This is a button that will handle form submission (we don't technically submit anything at this point but this button will take the form inputs and give us a result). If we use `<button type'submit'/>` the call to handleSubmit will be automatically handled. [see docs example](https://react-bootstrap.netlify.app/forms/overview/#overview).

Reply:
Thanks for your suggestion.
I've changed it.